File: ejercicios/AMarotoChivite/AMarotoChivite-01-10-Catan-Java/src/main/java/gameCatan/factories/BoxFactory.java
package gameCatan.factories;

import gameCatan.enums.TypeOwner;
import gameCatan.enums.TypeResource;
import gameCatan.models.Box;

import static gameCatan.utils.randomData.randomType;

public class BoxFactory {
    public static Box create() {
        TypeResource type = randomType();
        int randomValue = (int) (Math.random() * 6 + 1);
        return new Box(type, TypeOwner.NONE, randomValue);
    }
}

File: ejercicios/AMarotoChivite/AMarotoChivite-01-10-Catan-Kotlin/src/main/kotlin/gameCatan/factories/BoxFactory.kt
package gameCatan.factories

import gameCatan.enums.TypeOwner
import gameCatan.models.Box
import gameCatan.utils.randomData

object BoxFactory {
    @JvmStatic
    fun create(): Box {
        val type = randomData.randomType()
        val randomValue = (Math.random() * 6 + 1).toInt()
        return Box(type, TypeOwner.NONE, randomValue)
    }
}